source/pressure_tensor.cpp

```diff
@@ -0,0 +1,57 @@
+#include "pele/pressure_tensor.h"
+#include "pele/simple_pairwise_potential.h"
+
+namespace pele {
+
+/*
+ * computes the static pressure tensor, ignoring the momenta of the atoms
+ * the momentum component can be added
+ * */
+double pressure_tensor(std::shared_ptr<pele::BasePotential> pot_,
+                       pele::Array<double> x,
+                       pele::Array<double> ptensor,
+                       const double volume)
+{
+    pele::SimplePairwisePotentialInterface* pot = dynamic_cast<pele::SimplePairwisePotentialInterface*>(pot_.get());
+    if (pot == NULL) {
+        throw std::runtime_error("pressure_tensor: illegal potential");
+    }
+    const size_t ndim = pot->get_ndim();
+    const size_t natoms = x.size() / ndim;
+    if (ndim * natoms != x.size()) {
+        throw std::runtime_error("x is not divisible by the number of dimensions");
+    }
+    if (ptensor.size() != ndim * ndim) {
+        throw std::runtime_error("ptensor must have size ndim * ndim");
+    }
+    double gij;
+    double dr[ndim];
+    ptensor.assign(0.);
+    for (size_t atomi = 0; atomi < natoms; ++atomi) {
+        size_t const i1 = ndim * atomi;
+        for (size_t atomj = 0; atomj < atomi; ++atomj) {
+            size_t const j1 = ndim * atomj;
+            pot->get_rij(dr, &x[i1], &x[j1]);
+            double r2 = 0;
+            for (size_t k = 0; k < ndim; ++k) {
+                r2 += dr[k] * dr[k];
+            }
+            pot->get_interaction_energy_gradient(r2, &gij, atomi, atomj);
+            for (size_t k = 0; k < ndim; ++k) {
```

Comment:
OK, I think the summation over particle pairs was fine as it is. First I thought we were missing a factor 1/2 because some definitions were considering the vector pointing from the center of the particle to the contact (not to the other center), but then they sum over all particles and their neighbors (like Eq. 2 here: http://arxiv.org/pdf/1404.1854v1.pdf ), which to me sounds like taking each pair twice, so that the factor 1/2 goes away.
But concerning the matrix indices I was not sure. Would you agree that the latest version looks reasonable, given the definition in Allen-Tildesley? (The way it was done before would sum the diagonal twice, I believe.)
Concerning the sign of the vector, I changed it to the opposite now. I think that in pele
```
gij == - dV / dr / r
```
such that the force is gij * dr. Does that make sense? (Also the scalar pressure is positive then.)